### PR TITLE
fix: 🐛 use next/script for PublicEnvScript

### PR DESCRIPTION
**Pull request by davevanhoorn**
This PR:
- uses the `Script` tag from `next/script`, instead of the native HTML `script` tag in the `env-script.tsx` file
- it uses the `beforeInteractive` strategy, so runtime variables are available (`Load the script before any Next.js code and before any page hydration occurs.`)

Notes:
- We are using this package the same as people [here](https://github.com/expatfile/next-runtime-env/issues/107), and they've got the same hydration issues.

### CODE REVIEW

File: src/script/env-script.tsx [hunk 1]
```diff
@@ -35,9 +36,15 @@ export const EnvScript: FC<EnvScriptProps> = ({ env, nonce }) => {
     nonceString = nonce;
   }
 
+  const strategy =
+    process.env?.NEXT_RUNTIME_ENV === 'TEST' ||
+    process.env?.NEXT_PUBLIC_RUNTIME_ENV === 'TEST'
+      ? 'afterInteractive'
+      : 'beforeInteractive';
```
Comment (HofmannZ) [inline src/script/env-script.tsx:43]: Why don't we want `beforeInteractive` for tests?

File: src/script/env-script.spec.tsx [hunk 2]
```diff
@@ -35,17 +30,20 @@ describe('EnvProvider', () => {
...
     const nonce = 'test-nonce-xyz';
 
-    const { getByTestId } = render(<EnvScript env={env} nonce={nonce} />);
+    render(<EnvScript env={env} nonce={nonce} />);
 
-    expect(getByTestId('env-script')).toHaveAttribute('nonce', nonce);
+    expect(document.querySelector('script')).not.toHaveAttribute(
+      'nonce',
+      nonce,
+    );
```
Comment (HofmannZ) [inline src/script/env-script.spec.tsx:38]: This does not seem correct. We do want to see a nonce in this case.

File: src/script/env-script.spec.tsx [hunk 3]
```diff
@@ -4,29 +4,24 @@ import { render } from '@testing-library/react';
...
-describe('EnvProvider', () => {
   beforeEach(() => {
-    process.env = {};
+    jest.resetModules();
+    process.env = { NEXT_RUNTIME_ENV: 'TEST', ...processEnv };
+  });
+
+  afterAll(() => {
+    process.env = processEnv;
   });
```
Comment (HofmannZ) [inline src/script/env-script.spec.tsx:17]: For consistency with other tests, can you move this outside the describe block?

File: src/script/env-script.tsx (hunk 1 above)
Comment (davevanhoorn) [reply src/script/env-script.tsx:43]: The `Script` tag from Next doesn't seem to work in the tests (this is most likely a skill issue of me). I've now mocked the return of the `<script>` tag that we had before.

**Review by HofmannZ — APPROVED** [review]

**Pull request merged**